### PR TITLE
[Doc] Improve Python DataFrame documentation

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

**Comment by rxin** [comment]
@davies can you take a look at the changes?

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -894,7 +888,7 @@ def agg(self, *exprs):
 
     @dfapi
     def count(self):
-        """ Count the number of rows for each group.
+        """Counts the number of records for each group.
```
Comment (davies) [inline python/pyspark/sql/dataframe.py]: Add a line to say it will return a DataFrame.

**Comment by davies** [comment]
LGTM, just one minor comment.

**Comment by rxin** [comment]
I'm going to merge this and will fix the comment in a separate PR.

**Pull request closed without merging**